internal/controller/history.go

```diff
@@ -0,0 +1,54 @@
+package controllers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+)
+
+type History struct {
+	kuadrant *kuadrantv1beta1.Kuadrant
```

Comment:
This has a bit of a code smell and seems to indicate a possible issue in the design of the topology? Like if the pod crashes does this not mean history is rewritten from the point the pod comes back to be the current kuadrant state?